pkg/allocator/allocator.go

```diff
@@ -590,7 +591,15 @@ func (a *Allocator) lockedAllocate(ctx context.Context, key AllocatorKey) (idpoo
 	}
 
 	a.logger.Debug("Allocating new master ID", logfields.Key, k)
-	id, strID, unmaskedID := a.selectAvailableID()
+
+	// Hash the key. This is used as a hint so that agents
+	// tend to allocate the same numeric identity for the same labels.
+	// However, if this is the last attempt, then fall back to unseeded randomness
+	seed := crc64.Checksum([]byte(key.GetKey()), crc64.MakeTable(crc64.ECMA))
```

Comment:
This will now make the ID associated with any given set of labels somewhat predictable, which IIRC raised some concerns in the past as it may have consequences from the overall security posture point of view. Mostly raising the point just to see if further discussion on this may be appropriate.